icu4c/source/i18n/calendar.cpp

```diff
@@ -266,7 +266,7 @@ static ECalType getCalendarTypeForLocale(const char *locid) {
     //TODO: ULOC_FULL_NAME is out of date and too small..
     char canonicalName[256];
 
-    // canonicalize, so grandfathered variant will be transformed to keywords
+    // Canonicalize, so that an old-style variant will be transformed to keywords.
```

Comment:
Is _old-style variant_ really the right expression to use here, wouldn't these be _legacy variants_?

Reply:
"grandfathered" was misleading here. Variants like TRADITIONAL predate even our early ICU-syntax '@' key-value extension mechanism, not to mention BCP 47. In fact, the profusion of these kinds of orthogonal tweaks expressed as unusable variants drove us to inventing the '@' syntax first and then to add extensions to BCP 47.

I don't want to use "legacy" here because it would be misunderstood as code for what BCP 47 calls "grandfathered".